viewer/viewer-awt/src/main/java/org/icepdf/ri/common/utility/search/SearchFilterButtonWrapper.java

```diff
@@ -0,0 +1,226 @@
+            filterDropDownButton.addSeparator();
+            filterDropDownButton.add(showPagesCheckbox);
+        } else {
+            filterDropDownButton.add(wholeWordCheckbox);
+            filterDropDownButton.add(caseSensitiveCheckbox);
+            filterDropDownButton.add(commentsCheckbox);
+        }
+    }
+
+    public DropDownButton getButton() {
+        return filterDropDownButton;
+    }
+
+    public JCheckBoxMenuItem getWholeWordCheckbox() {
+        return wholeWordCheckbox;
+    }
+
+    public JCheckBoxMenuItem getRegexCheckbox() {
+        return regexCheckbox;
+    }
+
+    public JCheckBoxMenuItem getCaseSensitiveCheckbox() {
+        return caseSensitiveCheckbox;
+    }
+
+    public JCheckBoxMenuItem getCumulativeCheckbox() {
+        return cumulativeCheckbox;
+    }
+
+    public JCheckBoxMenuItem getTextCheckbox() {
+        return textCheckbox;
+    }
+
+    public JCheckBoxMenuItem getCommentsCheckbox() {
+        return commentsCheckbox;
+    }
+
+    public JCheckBoxMenuItem getDestinationsCheckbox() {
+        return destinationsCheckbox;
+    }
+
+    public JCheckBoxMenuItem getOutlinesCheckbox() {
+        return outlinesCheckbox;
+    }
+
+    public JCheckBoxMenuItem getShowPagesCheckbox() {
+        return showPagesCheckbox;
+    }
+
+    public boolean isWholeWord() {
+        return wholeWordCheckbox.isSelected();
+    }
+
+    public boolean isRegex() {
+        return regexCheckbox.isSelected();
+    }
+
+    public boolean isCaseSensitive() {
+        return caseSensitiveCheckbox.isSelected();
+    }
+
+    public boolean isCumulative() {
+        return cumulativeCheckbox.isSelected();
+    }
+
+    public boolean isText() {
+        return textCheckbox.isSelected();
+    }
+
+    public boolean isComments() {
+        return commentsCheckbox.isSelected();
+    }
+
+    public boolean isDestinations() {
+        return destinationsCheckbox.isSelected();
+    }
+
+    public boolean isOutlines() {
+        return outlinesCheckbox.isSelected();
+    }
+
+    public boolean isShowPages() {
+        return showPagesCheckbox.isSelected();
+    }
+
+    public void setEnabled(boolean b) {
+        filterDropDownButton.setEnabled(b);
+    }
+
+    public SearchTextTask getSearchTask(BaseSearchComponent panel, Controller controller, String pattern) {
+        SearchTextTask.Builder builder = new SearchTextTask.Builder(controller, pattern);
+        return builder.setSearchPanel(panel)
+                .setCaseSensitive(isCaseSensitive())
+                .setWholeWord(isWholeWord())
+                .setCumulative(isCumulative())
+                .setShowPages(isShowPages())
+                .setRegex(isRegex())
+                .setText(isText())
+                .setDestinations(isDestinations())
+                .setOutlines(isOutlines())
+                .setComments(isComments()).build();
+    }
+
+    public SimpleSearchHelper getSimpleSearchHelper(Controller controller, String pattern) {
```

Comment:
I really like the unification work that was done here between the two tool bars.